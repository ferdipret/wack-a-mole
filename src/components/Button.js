import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const ButtonWrapper = styled.button`
  background-color: white;
  border: 1px solid hsl(120, 50%, 70%);
  border-radius: 2px;
  color: hsl(120, 50%, 70%);
  cursor: pointer;
  display: flex;
  font-family: 'Share Tech Mono', monospace;
  font-size: 18px;
  justify-content: center;
  margin: 0 auto;
  padding: 10px;
  width: 120px;

  &:hover {
    border: 1px solid hsl(120, 50%, 60%);
    color: hsl(120, 50%, 60%);
  }

  &:focus {
    outline: none;
  }
`

function Button ({running, onClick, gameover}) {
  return (
    <ButtonWrapper onClick={onClick} className="main-btn">
      {gameover
        ? "RESTART"
        : running ? "PAUSE": "START"
      }
    </ButtonWrapper>
  )
}

Button.propTypes = {
  running: PropTypes.bool,
  onClick: PropTypes.func,
  gameover: PropTypes.bool,
}

export default Button
