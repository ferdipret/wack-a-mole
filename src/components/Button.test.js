import React from "react"
import { shallow } from "enzyme"
import Button from "./Button"

describe("button", () => {
  const wrapper = shallow(<Button />)
  it("renders properly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe("give some dynamic props", () => {
    it("should update the button text", () => {
      expect(wrapper.find(".main-btn").dive().text()).toEqual("START")

      wrapper.setProps({running: true})
      expect(wrapper.find(".main-btn").dive().text()).toEqual("PAUSE")

      wrapper.setProps({gameover: true})
      expect(wrapper.find(".main-btn").dive().text()).toEqual("RESTART")
    })
  })
})
