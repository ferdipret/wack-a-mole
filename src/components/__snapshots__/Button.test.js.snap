// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders properly 1`] = `
<styled.button
  className="main-btn"
>
  START
</styled.button>
`;
